lib/src/modules/macho/mod.rs

```diff
@@ -182,6 +182,42 @@ fn ep_for_arch_subtype(
     None
 }
 
+/// The function for checking if any dylib name present in the main Mach-O or embedded Mach-O files
+/// contain a dylib with the desired name
+///
+/// # Arguments
+///
+/// * `ctx`: A mutable reference to the scanning context.
```

Comment:
You don't need to document these functions with that level of details, with an "arguments"  and "returns" sections. Particularly the `ctx` shouldn't be included in the description.

The documentation for this function should be written from the perspective of someone using the function in YARA. The idea is that I can copy and paste the documentation for these exported functions and put it in the public documentation for the module. The `ctx` argument is an internal argument that is not visible from YARA.

Use the PE module as model of how to write the documentation for exported functions: https://github.com/VirusTotal/yara-x/blob/main/lib/src/modules/pe/mod.rs#L192-L203

Reply:
ack, thanks Victor will rework the doc comments for the functions 😄

Reply:
updated all the functions i've added 😸